rust/datafusion/src/logical_plan/expr.rs

```diff
@@ -160,20 +160,26 @@ pub enum Expr {
     },
     /// Represents the call of a built-in scalar function with a set of arguments.
     ScalarFunction {
+        /// The input name of the function
+        input_name: String,
```

Comment:
Also, I looked through the diff and I couldn't figure out what `input_name` is actually used for. Is this new field needed?

Reply:
This is mainly used to display the name of the function entered by the user. 
md5 (Utf8("a")) instead of mD5(Utf8("a")) if there is no input_name.

For example:

> mysql> select mD5('a');
> +----------------------------------+
> | mD5(Utf8("a"))                   |
> +----------------------------------+
> | 0cc175b9c0f1b6a831c399e269772661 |
> +----------------------------------+
> 1 row in set (0.01 sec)

Reply:
FWIW here is what postgres does for the same query (uses the lowercase, canonical function name)

```
alamb=# select mD5('a');
               md5                
----------------------------------
 0cc175b9c0f1b6a831c399e269772661
(1 row)
```